File: push/src/main/java/top/trumeet/mipushframework/wizard/permission/AlertWindowPermissionInfo.java
package top.trumeet.mipushframework.wizard.permission;

import android.content.Context;
import android.os.Build;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;

import com.xiaomi.xmsf.R;

@RequiresApi(api = Build.VERSION_CODES.M)
public class AlertWindowPermissionInfo implements PermissionInfo {
    private final Context context;

    public AlertWindowPermissionInfo(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public PermissionOperator getPermissionOperator() {
        return new AlertWindowPermissionOperator(context);
    }

    @Override
    @NonNull
    public String getPermissionTitle() {
        return context.getString(R.string.wizard_title_alert_window_permission);
    }

    @Override
    @NonNull
    public String getPermissionDescription() {
        return context.getString(R.string.wizard_title_alert_window_text);
    }
}

File: push/src/main/java/top/trumeet/mipushframework/wizard/permission/AlertWindowPermissionInfo.kt
package top.trumeet.mipushframework.wizard.permission

import android.content.Context
import android.os.Build
import androidx.annotation.RequiresApi
import com.xiaomi.xmsf.R

@RequiresApi(api = Build.VERSION_CODES.M)
class AlertWindowPermissionInfo(private val context: Context) : PermissionInfo {
    override val permissionOperator: PermissionOperator
        get() = AlertWindowPermissionOperator(context)

    override val permissionTitle: String
        get() = context.getString(R.string.wizard_title_alert_window_permission)

    override val permissionDescription: String
        get() = context.getString(R.string.wizard_title_alert_window_text)
}
